crates/burn-wgpu/src/compiler/wgsl/compiler.rs

```diff
@@ -129,6 +135,7 @@ impl<F: FloatElement, I: IntElement> WgslCompiler<F, I> {
     fn compile_elem(value: gpu::Elem) -> wgsl::Elem {
         match value {
             gpu::Elem::Float => F::wgpu_elem(),
+            gpu::Elem::Half => F::wgpu_elem(),
```

Comment:
This line pretty much explains why we don't need `Half` in the `gpu::Elem` enum :)